models/postgres/txcontext.go

```diff
@@ -0,0 +1,53 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+)
+
+type contextKey int
+
+var contextKeyTx = contextKey(0)
+
+type TxContext struct {
+	*sql.DB
+}
+
+func (t *TxContext) WithTxContext(ctx context.Context, f func(context.Context) error) error {
+	tx, err := t.Begin()
+	if err != nil {
+		return err
+	}
+
+	ctx = context.WithValue(ctx, contextKeyTx, tx)
+
+	err = f(ctx)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	return nil
+}
+
+type Tx interface {
```

Comment:
Can we directly inherit sql.DB here?

Reply:
`DB` is a struct. We need an interface here.